spec/faraday/middleware_registry_spec.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class CustomMiddleware < Faraday::Middleware
+end
```

Comment:
Perhaps this can become something like

```ruby
let(:custom_middleware_klass) { Class.new(Faraday::Middleware) }
```

Reply:
I originally went with that, but not sure how to make this work with the tests since they would not be able to find the class. Do you know if there's a way to make them pass doing this?

Reply:
This made it pass!

```ruby
# frozen_string_literal: true

RSpec.describe Faraday::MiddlewareRegistry do
  before do
    stub_const("CustomMiddleware", custom_middleware_klass)
  end
  let(:custom_middleware_klass) { Class.new(Faraday::Middleware) }
  let(:dummy) { Class.new { extend Faraday::MiddlewareRegistry } }

  after { dummy.unregister_middleware(:custom) }

  it 'allows to register with constant' do
    dummy.register_middleware(custom: custom_middleware_klass)
    expect(dummy.lookup_middleware(:custom)).to eq(custom_middleware_klass)
  end

  it 'allows to register with symbol' do
    dummy.register_middleware(custom: :CustomMiddleware)
    expect(dummy.lookup_middleware(:custom)).to eq(custom_middleware_klass)
  end

  it 'allows to register with string' do
    dummy.register_middleware(custom: 'CustomMiddleware')
    expect(dummy.lookup_middleware(:custom)).to eq(custom_middleware_klass)
  end

  it 'allows to register with Proc' do
    dummy.register_middleware(custom: -> { custom_middleware_klass })
    expect(dummy.lookup_middleware(:custom)).to eq(custom_middleware_klass)
  end
end
```

Reply:
If you want me to, I could push that to this PR.

Reply:
Fantastic! TIL `stub_const` 😄!
Yes please, feel free to push and merge as I'm stuck in meetings for another hour or so 😭

Reply:
Alright, took me long enough, but now I found your comment, and have pushed the change.

Reply:
I resolved the conflict I had introduced re: CHANGELOG, force-pushed.